src/hooks/useColonySubscription.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import {
+  GetFullColonyByNameDocument,
+  useCreateWatchedColoniesMutation,
+  useDeleteWatchedColoniesMutation,
+} from '~gql';
+import { CREATE_USER_ROUTE } from '~routes';
+import { useAppContext, useColonyContext } from '~hooks';
+import { getWatchedColony } from '~utils/watching';
+import { handleNewUser } from '~utils/newUser';
+
+const useColonySubscription = () => {
+  const { colony, canInteractWithColony } = useColonyContext();
+  const {
+    user,
+    updateUser,
+    wallet,
+    walletConnecting,
+    connectWallet,
+    userLoading,
+  } = useAppContext();
+  const navigate = useNavigate();
+
+  const watchedItem = getWatchedColony(colony, user?.watchlist?.items);
+
+  /* Watch (follow) a colony */
+  const [subscribe, { data: watchData }] = useCreateWatchedColoniesMutation({
```

Comment:
nitpick, but subscribe may be misleading since it's not actually a subscription in the apollo sense of the term. I think `watch()` \ `unwatch()` captures the same intent?